src/ModalFooter.js

```diff
@@ -2,30 +2,24 @@ import classNames from 'classnames';
 import React from 'react';
 import elementType from 'prop-types-extra/lib/elementType';
 
-import { bsClass, getClassSet, splitBsProps } from './utils/bootstrapUtils';
+import { createBootstrapComponent } from './ThemeProvider';
 
-const propTypes = {
-  componentClass: elementType
-};
+class ModalFooter extends React.Component {
+  static propTypes = {
+    componentClass: elementType
+  };
 
-const defaultProps = {
-  componentClass: 'div'
-};
+  static defaultProps = {
+    componentClass: 'div'
+  };
 
-class ModalFooter extends React.Component {
   render() {
     const { componentClass: Component, className, ...props } = this.props;
-    const [bsProps, elementProps] = splitBsProps(props);
 
-    const classes = getClassSet(bsProps);
+    const classes = classNames('modal-footer', className);
```

Comment:
So `createBootstrapComponent` will inject the string you specified in as a `bsPrefix` prop, in this case it'll be `modal-footer`. That let's you not hardcode the `modal-footer` bit here, and instead for `className(className, bsPrefix)` to add the class. For cases where there are variants (like dialog), you can use both the plain bsPrefix, and then intropolated string to create all the classes like, ```classNames(bsPrefix, `${bsPrefix}-content`)```